lib/htmllint.js

```diff
@@ -23,6 +23,21 @@ module.exports = function( config, done ) {
     return str;
   }
 
+  // parse and, if needed, normalize error messages from HttpClient to vnu.jar format
+  function parseErrorMessages( errors ) {
+    var parsed = JSON.parse( config.server ? '[' + errors.trim().replace( /\n/g, ',' ) + ']' : errors );
+    var messages = parsed.messages;
+    if ( config.server ) {
+      messages = [].concat.apply([], parsed.map(function( file ) {
```

Comment:
Gotcha, thanks for the clarification! Can you add a source comment about the "resulting in an array of arrays, instead of an array of objects, which is then flattened by the concatenation." part? That seems key here.